In [40]:
import pandas as pd
import numpy as np
from math import sqrt

### Задание 1
Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [11]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [13]:
# Находим ковариацию с помощью формулы
cov = np.mean(zp*ks) - np.mean(zp) * np.mean(ks)
cov

9157.839999999997

In [19]:
# ковариация с помощью numpy
np.cov(zp,ks,ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [34]:
# Находим коэффициэнт корреляции Пирсона по формуле
sigma_zp = np.sqrt(((zp - zp.mean())**2).sum() / len(zp))
sigma_ks = np.sqrt(((ks - ks.mean())**2).sum() / len(ks))

cov / (sigma_zp*sigma_ks)

0.8874900920739158

In [24]:
# Коэффициэнет корреляции с помощью numpy
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

### Задание 2
Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [50]:
iq = [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
df = pd.DataFrame({'iq':iq})
n = len(iq) #10
p = 0.95
a = 0.05
MX = df['iq'].mean()
MX

118.1

In [36]:
# Находим несмещенную дисперсию
D = df['iq'].var(ddof=1)
D

111.21111111111111

In [41]:
# среднее квадратичное несмещенной дисперсии
s = sqrt(D)
s

10.54566788359614

Поскольку сигма неизвестна, используем параметр t в формуле $$\overline{X} \pm t_{таб}\frac{s}{\sqrt{n}}$$

Находим t по таблице с параметрами (n-1;α/2) = (9;0.025) = 2.262

In [51]:
X1 = MX - 2.262 * (s /sqrt(n))
X1

110.55660776308164

In [52]:
X2 = MX + 2.262 * (s /sqrt(n))
X2

125.64339223691834

#### Доверительный интервал для математического ожидания: [110.5566; 125.6434]

### Задание 3
Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [49]:
D = 25
n = 27
MX = 174.2
p = 0.95
a = 0.05

In [54]:
sigma = sqrt(D)

In [56]:
# Сигма известна, используем параметр Z. По таблице с параметрами (1 - a/2 = 0.975) = 1.96
z = 1.96
X1 = MX - (z*sigma/sqrt(n))
X1

115.00096789303498

In [57]:
X2 = MX + (z*sigma/sqrt(n))
X2

121.199032106965

#### Доверительный интервал для математического ожидания: [115.0097; 121.1990]